Issue 73: Add Debug Logging

maupind opened the pull request:
Added debug logging as per issue 73 for more verbose error messages. I added to all files except for `trajectory_gpt2.py` as this already had logging imported from `transformers.utils`. 

I kept the `logger.debug(f'foobar')` as an example of usage, but happy to update this to have more file specific message if required.

Commit:
add debug logging

Commit:
add debug logging full

eihli commented:
Anyone have any thoughts on what we should do with all of the existing calls to `print(..)`? I was thinking we could update them to `log.info` for now. `info` seems like a reasonable default level. And if anything jumps out as obviously debug or obviously error, then `log.<level>` appropriately.

eihli commented:
TODO: I'm curious if we'll run into anything interesting when it comes to [Accelerate](https://github.com/huggingface/accelerate) and logging from [multiple processes](https://docs.python.org/3/howto/logging-cookbook.html#logging-to-a-single-file-from-multiple-processes).

eihli committed:
Update print to logger.info

Also removes logging imports from files that don't do any logging
and removes the example comment. Both of those changes are just minor
non-functional changes to reduce the number of lines of non-functional code.
The logging module is standard to python and there are plenty of examples
in the official Python docs, so I don't think we gain much by including
examples in each file.

eihli commented:
@maupind check out this diff and let me know what you think: https://github.com/maupind/NEKO/pull/1

maupind commented:
@eihli Thanks for the update. I had meant to respond your comment on what to with the existing print statements, but got a little busy! The changes all seemed fine to me, but I am not an expert in logging.

Regarding the interaction with Accelerate and logging from multiple processes, is this something we can test through?

eihli committed:
Refactor to use prints in some places

Trying to differentiate between logging and printing. Logging is for informational
and debug text. Printing is like a GUI. Printing is how you interact with the program
on every run. That's how I'm thinking about it in my head right now.

maupind committed:
Merge pull request #3 from eihli/maupind/issue-73

Fix issue with previous PR. I should have made the PR into your issue branch rather than your main branch.

maupind committed:
Merge branch 'master' into issue_73_add_debug_logging

maupind commented:
Apologies @eihli, I was away on leave and missed your previous request. Is there anything else required at this stage?
